### PR TITLE
CCUBE-1281

### PR DESCRIPTION
**Pull request by ShIrInRuTh**
**Changes**
- Enabled prefill inputs to be saved via RHF.
- Added validation schema for prefill fields.
- Added controller for prefill fields.
- Added error handling for the prefill fields.
- Added isDirty checks and useEffects to change form to isDirty when form is touched.
- Amended tests for prefill and prefill child to add form provider tags.
- Added tests to prefill child for testing of required field error handling & invalid input error handling.
- Amended test variables and names.

**Additional information**

-   You may refer to this [CCUBE-1281](https://jira.ship.gov.sg/browse/CCUBE-1281)
-   Any other information you would like to include

### CODE REVIEW

File: src/components/common/multi-entry/multi-entry.styles.tsx [hunk 1]
```diff
@@ -1,9 +1,20 @@
+import { Text } from "@lifesg/react-design-system/text";
 import { Accordion } from "@lifesg/react-design-system/accordion";
 import { ButtonWithIcon } from "@lifesg/react-design-system/button-with-icon";
 import { Color } from "@lifesg/react-design-system/color";
 import styled from "styled-components";
 
-export const MultiEntryAccordionItem = styled(Accordion.Item)`
+interface MultiEntryAccordionItemProps {
```
Comment (keithtxw) [inline src/components/common/multi-entry/multi-entry.styles.tsx]: Add section header `STYLE INTERFACE`

Comment (keithtxw) [inline src/components/common/multi-entry/multi-entry.styles.tsx]:
```suggestion
interface IMultiEntryAccordionItemStyleProps {
```

File: src/context-providers/builder/element.types.ts [hunk 2]
```diff
@@ -34,6 +34,12 @@ export interface IConditionalRendering {
     internalId: string;
 }
 
+export interface IPrefill {
```
Comment (keithtxw) [inline src/context-providers/builder/element.types.ts]:
```suggestion
export interface IPrefillAttributes {
```

File: src/components/element-editor/prefill/prefill.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+
+    const handleChildChange = (index: number, newValue: IPrefill) => {
+        setChildEntryValues((prevValues) => {
+            const updatedValues = [...prevValues];
+            updatedValues[index] = newValue;
+            setValue("prefill", updatedValues);
```
Comment (keithtxw) [inline src/components/element-editor/prefill/prefill.tsx:30]: You already have an effect function that updates already. There is no need to redo it here.

File: src/components/element-editor/prefill/prefill.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    const handleDelete = (index: number) => {
+        setChildEntryValues((prevValues) => {
+            const updatedValues = prevValues.filter((_, i) => i !== index);
+            setValue("prefill", updatedValues);
+            return updatedValues;
+        });
+    };
+
+    // =========================================================================
+    // USE EFFECTS
```
Comment (keithtxw) [inline src/components/element-editor/prefill/prefill.tsx]:
```suggestion
    // EFFECTS
```

File: src/components/element-editor/prefill/prefill.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            updateFocusedElement(true);
+        }
+    }, [isDirty, updateFocusedElement]);
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+
+    const renderChildren = () => {
+        return childEntryValues?.map((child, index) => (
```
Comment (keithtxw) [inline src/components/element-editor/prefill/prefill.tsx]: have a check, if `childEntryValues` is an empty array, then don't iterate and render

File: src/context-providers/builder/element.types.ts [hunk 6]
```diff
@@ -34,6 +34,12 @@ export interface IConditionalRendering {
...
+    prefillMode?: string;
```
Comment (keithtxw) [inline src/context-providers/builder/element.types.ts]: Should this be some string literal? What are the values?

**Review by keithtxw — DISMISSED** [review]

File: src/components/element-editor/prefill/prefill.tsx (hunk 3 above)
Comment (ShIrInRuTh) [reply src/components/element-editor/prefill/prefill.tsx:30]: if i remove line 28, the updating of the fields is a little slow and it does this 😅
image

File: src/context-providers/builder/element.types.ts (hunk 6 above)
Comment (ShIrInRuTh) [reply src/context-providers/builder/element.types.ts]: act the value for this is either 'Myinfo' or 'Previous source'

File: src/components/common/multi-entry/multi-entry.styles.tsx [hunk 7]
```diff
@@ -1,9 +1,23 @@
...
 
-export const MultiEntryAccordionItem = styled(Accordion.Item)`
+// =============================================================================
+// STYLE INTERFACE
+// =============================================================================
+interface IMultiEntryAccordionItemStyleProps {
+    $hasSubtitle: boolean;
+}
+
+export const SubtitleText = styled(Text.Body)`
```
Comment (weili-govtech) [inline src/components/common/multi-entry/multi-entry.styles.tsx:18]: now that we have a divider for the interface, we can have another divider for the stylings

```suggestion
// =============================================================================
// STYLING
// =============================================================================

export const SubtitleText = styled(Text.Body)`
```

File: src/context-providers/builder/element.types.ts (hunk 6 above)
Comment (weili-govtech) [reply src/context-providers/builder/element.types.ts]: let's go for string literal, in the long run, we will go with discriminating union

```suggestion
    prefillMode?:"Myinfo"|"Previous source";
```

**Review by weili-govtech — CHANGES_REQUESTED** [review]

File: src/components/element-editor/prefill/prefill.tsx (hunk 3 above)
Comment (keithtxw) [reply src/components/element-editor/prefill/prefill.tsx:30]: Hmm alright then. It feels a little weird to update the context in a state update. But i guess that is fine if it is to solve a specific issue.

**Review by weili-govtech — APPROVED** [review]

**Pull request merged**